src/rendergraph/opengl/backend/basenode.cpp

```diff
@@ -0,0 +1,59 @@
+#include "backend/basenode.h"
+
+#include "rendergraph/assert.h"
+#include "rendergraph/engine.h"
+
+using namespace rendergraph;
+
+BaseNode::~BaseNode() {
+    DEBUG_ASSERT(m_pParent == nullptr);
+    DEBUG_ASSERT(m_pEngine == nullptr);
+
+    while (m_pFirstChild) {
+        std::unique_ptr<BaseNode> pChild(m_pFirstChild);
+        removeChildNode(pChild.get());
```

Comment:
this just looks like a verbose way of saying `delete`.

Reply:
Yes. But it nicely expresses that the pointer has to be deleted at the end of the scope.

Anyway, I don’t feel strong about this, I’ll happily change this to a delete if you prefer.

Reply:
I think so yeah. Otherwise the `unique_ptr` just hides that there is manual memory management going on. If we're doing manual memory management, I would like to see the raw `new` and `delete`s so it raises the proper alarm bells when reading the code.

Reply:
Ok. I’ll change it. Note there is ideally no new, as I provide an interface for appendChildNode that uses unique_ptr (and release() internally)

Reply:
that would be great. Why `release()` though? Can't we keep it managed? If not consider looking into `gsl::owner` to mark the pointer accordingly (since according to the cppcoreguidelines, raw pointers should be assumed to be non-owning).

Reply:
We could, but Qt uses raw pointers for the node hierarchy, and here it really is much simpler to mimick the scenegraph API in the opengl layer. I had this different earlier but the resulting code was much more confusing. Anyway, it is isolated, internal and ownership is clear.

Reply:
I added comments.

Reply:
Please re-review and resolve conversation.

Reply:
I'll look into this some other time, I'll allow the raw pointers if thats the basically the sole responsibility of the class.